src/wasm-lib/kcl/src/std/shell.rs

```diff
@@ -148,27 +148,67 @@ pub async fn shell(exec_state: &mut ExecState, args: Args) -> Result<KclValue, K
 /// // We put "thing1" in the shell function to shell the end face of the object.
 /// shell({ faces: ['end'], thickness: 5 }, thing1)
 /// ```
+///
+/// ```no_run
+/// // Shell sketched on face objects on the end face, include all sketches to shell
+/// // the entire object.
+///
+/// let size = 100
+/// const case = startSketchOn('XY')
+///     |> startProfileAt([-size, -size], %)
+///     |> line([2 * size, 0], %)
+///     |> line([0, 2 * size], %)
+///     |> tangentialArcTo([-size, size], %)
+///     |> close(%)
+///     |> extrude(65, %)
+///
+/// const thing1 = startSketchOn(case, 'end')
+///     |> circle([-size / 2, -size / 2], 25, %)
+///     |> extrude(50, %)
+///
+/// const thing2 = startSketchOn(case, 'end')
+///     |> circle([size / 2, -size / 2], 25, %)
+///     |> extrude(50, %)
+///
+/// // We put "thing1" and "thing2" in the shell function to shell the end face of the object.
+/// shell({ faces: ['end'], thickness: 5 }, [thing1, thing2])
+/// ```
 #[stdlib {
     name = "shell",
 }]
 async fn inner_shell(
     data: ShellData,
-    extrude_group: Box<ExtrudeGroup>,
+    extrude_group_set: ExtrudeGroupSet,
     exec_state: &mut ExecState,
     args: Args,
-) -> Result<Box<ExtrudeGroup>, KclError> {
+) -> Result<ExtrudeGroupSet, KclError> {
     if data.faces.is_empty() {
         return Err(KclError::Type(KclErrorDetails {
             message: "Expected at least one face".to_string(),
             source_ranges: vec![args.source_range],
         }));
     }
 
+    let extrude_groups: Vec<Box<ExtrudeGroup>> = extrude_group_set.clone().into();
```

Comment:
Do we need to clone here?

Reply:
Hmm, I think it's possible, but I see that it's a bit annoying to iterate over them without cloning into a Vec.